### PR TITLE
Added missing types in Gradio UI handlers

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Added missing types in Gradio UI handlers

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Minor

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: ols/src/ui/gradio_ui.py [hunk 1]
```diff
@@ -14,8 +15,8 @@ class GradioUI:
 
     def __init__(
         self,
-        ols_url="http://127.0.0.1:8080/v1/query",
-        conversation_id=None,
+        ols_url: str = "http://127.0.0.1:8080/v1/query",
+        conversation_id: Optional[int] = None,
```
Comment (onmete) [inline ols/src/ui/gradio_ui.py:19]: Shouldn't it be a `str`?

Comment (tisnik) [reply ols/src/ui/gradio_ui.py]: yes, and I had to fix unit test too

File: ols/src/ui/gradio_ui.py [hunk 2]
```diff
@@ -14,8 +15,8 @@ class GradioUI:
 
     def __init__(
         self,
-        ols_url="http://127.0.0.1:8080/v1/query",
-        conversation_id=None,
+        ols_url: str = "http://127.0.0.1:8080/v1/query",
+        conversation_id: Optional[str] = None,
```
Comment (bparees) [inline ols/src/ui/gradio_ui.py:19]: I think we started standardizing on using the `str | None = None` syntax for this, which is the python 3.11 style?

File: ols/src/ui/gradio_ui.py [hunk 3]
```diff
@@ -26,7 +27,7 @@ def __init__(
         use_history = gr.Checkbox(value=True, label="Use history")
         self.ui = gr.ChatInterface(self.chat_ui, additional_inputs=[use_history])
 
-    def chat_ui(self, prompt, history, use_history=None):
+    def chat_ui(self, prompt: str, history, use_history: Optional[bool] = None) -> str:
```
Comment (bparees) [inline ols/src/ui/gradio_ui.py:30]: `bool | None = None` ?

File: ols/src/ui/gradio_ui.py (hunk 2 above)
Comment (tisnik) [reply ols/src/ui/gradio_ui.py:19]: ok will change it.

File: ols/src/ui/gradio_ui.py (hunk 3 above)
Comment (tisnik) [reply ols/src/ui/gradio_ui.py]: updated

**Pull request merged**